tests/check/check_signed_video_sign.c

```diff
@@ -721,81 +721,6 @@ START_TEST(golden_sei_created)
 }
 END_TEST
 
-/* Test description
- * Generates SEIs using golden SEI prinsiple and verifies them.
- * The operation is as follows:
- * 1. Setup a signed_video_t session
- * 2. Generate and store golden SEI
- * 3. End signed_video_t and start a new one
- * 4. Enable golden SEI principle
- * 5. Introduce a I frame to generate a new SEI frame
- * 6. Get the SEI frame
- * 7. Verify the stored golden SEI and newly fetched SEI
- */
-START_TEST(golden_sei_principle)
-{
-
-  SignedVideoCodec codec = settings[_i].codec;
-  SignedVideoReturnCode sv_rc;
-  signed_video_t *sv = signed_video_create(codec);
-  ck_assert(sv);
-  char *private_key = NULL;
-  size_t private_key_size = 0;
-  // Setup the key
-  sv_rc = settings[_i].generate_key(NULL, &private_key, &private_key_size);
-  ck_assert_int_eq(sv_rc, SV_OK);
-
-  sv_rc = signed_video_set_private_key_new(sv, private_key, private_key_size);
-  ck_assert_int_eq(sv_rc, SV_OK);
-  sv_rc = signed_video_set_hash_algo(sv, settings[_i].hash_algo_name);
-  ck_assert_int_eq(sv_rc, SV_OK);
-  sv_rc = signed_video_generate_golden_sei(sv);
-  ck_assert_int_eq(sv_rc, SV_OK);
-
-  size_t golden_sei_size = 0;
-  sv_rc = signed_video_get_sei(sv, NULL, &golden_sei_size);
-  ck_assert(golden_sei_size != 0);
-  uint8_t *golden_sei = malloc(golden_sei_size);
-  ck_assert_int_eq(sv_rc, SV_OK);
-  // Store the |golden_sei|
-  sv_rc = signed_video_get_sei(sv, golden_sei, &golden_sei_size);
-  ck_assert_int_eq(sv_rc, SV_OK);
-
-  // End the |sv| session
-  signed_video_free(sv);
-  // Start a new stream
-  sv = signed_video_create(codec);
-  ck_assert(sv);
-  sv_rc = signed_video_set_using_golden_sei(sv, true);
```

Comment:
You should still test that this API fulfills what it says in the public description, that is, returns the correct error codes and cannot be used after starting a stream.